crypto/sha/sha3.c

```diff
@@ -8,13 +8,15 @@
  */
 
 #include <string.h>
+#include "crypto/s390x_arch.h"
 #include "internal/sha3.h"
 
 void SHA3_squeeze(uint64_t A[5][5], unsigned char *out, size_t len, size_t r, int next);
 
 void ossl_sha3_reset(KECCAK1600_CTX *ctx)
 {
-    memset(ctx->A, 0, sizeof(ctx->A));
+    if (!(OPENSSL_s390xcap_P.stfle[1] & S390X_CAPBIT(S390X_MSA12)))
```

Comment:
This needs to go into a `#if` so that it is only done on the s390x platform.

Reply:
S390_SHA3 is never defined in this source module. 
Better use `#if defined(OPENSSL_CPUID_OBJ) && defined(__s390__)`